Function support for aREST backend

bachp opened the pull request:
The implementation sends /<name>?params=1 for turn_on and /<name>?params=0 for turn_off
It uses the return value of the function to determine the current state.
0=Off, 1=On

bachp commented:
Update PR to fix pylint warning found by travis.

balloob commented on homeassistant/components/switch/arest.py:
```diff
@@ -34,30 +34,32 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         return False
```
Can you revert the file mode change for this file ?

bachp replied:
Ok I changed it back.

balloob commented on homeassistant/components/switch/arest.py:
```diff
@@ -68,6 +70,69 @@ def is_on(self):
+            _LOGGER.error("Can't turn off function %s at %s. Is device offline?",
+                          self._func, self._resource)
+
+    def update(self):
```
It looks like this one is similar to the pin based one. Can you set the url in the constructor and move this method to the base class?

If you define `turn_on_params` and `turn_off_params` in the constructor too, you'll be able to re-use `turn_on` and `turn_off` too.

bachp replied:
I'm not sure if this is better, More things would have to parametrize the "url path", the "params" and the "log message" too, even the `_pin` and `_func` attribute would need to be aligned.

It feels a bit like to much generalization too me. But I can do it if you want.

balloob replied:
Yeah, good point.

balloob commented:
You've got some [style issues](https://travis-ci.org/balloob/home-assistant/builds/93010495#L298-L302). For the rest, looks good.

bachp commented:
How strict is the 80 characters per line rule? I would have to break up the log string to fix the remaining two style issues.

fabaff commented:
To be precise, the limit is 79 characters according to PEP8 and `pylint` will continue to fail. So, for `pylint` this rule is strict. Breaking up the log messages is the way to go ;-)

bachp committed:
function support for aREST backend

The implementation sends /<name>?params=1 for turn_on and /<name>?params=0 for turn_off
It uses the return value of the function to determine the current state.
0=Off, 1=On

bachp commented:
Style changes applied
